### ISSUE TITLE
Module msn Not Found

### ISSUE THREAD

**Issue by alunduil** [taggartaa/calamity#3]
Here's the current problems I see with the code:

python calamity.py -h
Traceback (most recent call last):
  File "calamity.py", line 25, in <module>
    from calamity import Calamity, CalamityException
  File "/home/alunduil/programs/calamity/calamity/**init**.py", line 13, in <module>
    from calamity import Calamity, CalamityException
  File "/home/alunduil/programs/calamity/calamity/calamity.py", line 12, in <module>
    import network
  File "/home/alunduil/programs/calamity/calamity/network/**init**.py", line 12, in <module>
    from login import login
  File "/home/alunduil/programs/calamity/calamity/network/login.py", line 11, in <module>
    import msn
ImportError: No module named msn

**Comment by taggartaa** [comment]
Sorry it is taking so long to get back to you. The problems are coming from having the gui and network packages inside the calamity package (they are one directory level up on my end). I contemplated this difference, and I can see putting network inside calamity because it is written specifically for calamity, but gui is independent so it should definitely be on the outside. Do you want me to put the network package inside calamity because that does make some sense.

**Comment by alunduil** [comment]
The way I see it if you write generic libraries that are for more than this project they should get their own project and then be created as a submodulee in vendor (as I've done with pycolorize or will do I'll check in a moment).  Anything else should got in a calamity module so you don't pollute the global module space.  When these packages get installed all your modules will simply go in /usr/share/python2.6/site-packages (as is!).  Properly organizing the code is paramount to make sure that strange behavior is not experienced by any user (no matter how different their setup is from yours).

To recap:

All self-contained modules should be their own project and a submodule in vendor.
All calamity libraries should be in the calamity module.

**Comment by alunduil** [comment]
Also, an FYI.  I noticed this package in portage:
- net-im/pymsn-t
   Available versions:  (~)0.11.3 0.11.3-r1 0.11.3-r2
   Homepage:            http://msn-transport.jabberstudio.org/
   Description:         Python based jabber transport for MSN

Perhaps we shouldn't reinvent the wheel like good developers and just use this project?

**Comment by taggartaa** [comment]
That is a dead link. But I get what you're saying, I googled it and it seems I'd be pretty dumb to waste my time making a low level python interface for msn (or any other server for that matter). Good thinking. Google came back with msnp,py, I will look into it this weekend.

As for the gui thing, I hadn't thought of making it a separate project but that makes a lot of sense as well. I will start a new project for that and move the new network module inside calamity.

**Comment by alunduil** [comment]
I still think your abstract GUI factory should be a part of this project and just be a translator for pygtk and pyQt as well as pytk if it exists.  If the latter doesn't exist then I would recommend make a project for it.

**Comment by alunduil** [comment]
Found another package you may want to consider (it's the one ubuntu uses for empathy, I'm guessing). 
- dev-python/papyon
   Available versions:  0.4.6 (~)0.5.1 (~)0.5.1-r1 (~)0.5.2
   Homepage:            http://www.freedesktop.org/wiki/Software/papyon
   Description:         Python MSN IM protocol implementation forked from pymsn
